packages/shared/src/components/squads/SquadPageHeader.tsx

```diff
@@ -39,6 +59,30 @@ export function SquadPageHeader({
     tourIndex === TourScreenIndex.Post ||
     (isChecklistVisible && openStep === ActionType.SquadFirstPost);
   const isSquadMember = !!squad.currentMember;
+  const isFeatured = squad?.flags?.featured;
+
+  const buttonCopy = (() => {
+    if (isFeatured) {
+      return 'Featured';
+    }
+
+    return squad.public ? 'Public' : 'Private';
+  })();
+
+  const buttonIcon = (() => {
+    if (isFeatured) {
+      return <SourceIcon secondary />;
+    }
+
+    return squad.public ? <EarthIcon /> : <LockIcon />;
+  })();
```

Comment:
Maybe they can be done in one func and return as an object {copy, icon}?

Reply:
I was assuming a state with useEffect could work, but it's ok either way

Reply:
Ah, wait, we can maybe use a single object, let me check. However, we will probably avoid adding a `useEffect`.

Reply:
Updated to use a single object like you suggested. Thank you! 🥳